gustaf/utils/tictoc.py

```diff
@@ -2,3 +2,116 @@
 
 Timer that tics, tocs and logs.
 """
+from time import perf_counter as now
+
+from gustaf._base import GustafBase
+
+
+class Tic(GustafBase):
+    """
+    Timer class for easier time measurement.
+    """
+
+    __slots__ = ("_title", "_names", "_laps", "_logger")
+
+    # add short cut to perf_counter in case you just want pure "now"
+    now = now
+
+    def __init__(self, title="untitled", log_level="debug"):
+        """
+        Configures the timer and logger. Marks the starting point.
+
+        Parameters
+        ----------
+        title: str
+          title for this measurement.
+        log_level: str
+          Valid options are {"info", "debug", "warning"}
+        """
+        # select logger based on logger level
+        self._logger = eval(f"self._log{str(log_level).lower()[0]}")
+
+        # initialize names
+        self._names = []
+
+        # give name
+        self._title = str(title)
+
+        # start
+        self._laps = [now()]
+
+    def toc(self, name=None, log=False):
+        """
+        Adds now to the measurements.
+
+        Parameters
+        ----------
+        name: str
+          name of this specific measurement.
+          By default it assign a number to this lap.
+        log: bool
+          If True, will log lapsed time.
+
+        Returns
+        -------
+        None
+        """
+        # measure now
+        self._laps.append(now())
+
+        self._names.append(
+            f"lap-{len(self._laps) - 2}" if name is None else str(name)
+        )
+
+        if log:
+            self._logger("time lapsed: ", self._laps[-2] - self._laps[-1])
+
+    def summary(self, print_=False):
```

Comment:
Maybe also add an option of getting the data without having the log file filled? Maybe, a Getter function for the lap times?

def timeings(self)->List[Tuple[str, time]] or something like this.